parser/internal/names/scope.go

```diff
@@ -0,0 +1,37 @@
+package names
```

Comment:
This type was just extracted from the `resolvers.go` file to make the code easier to read; otherwise I made no changes here